src/components/OptionsSelector/BaseOptionsSelector.js

```diff
@@ -144,13 +144,19 @@ class BaseOptionsSelector extends Component {
      */
     flattenSections() {
         const allOptions = [];
+        this.disabledOptionsIndexes = [];
```

Comment:
any specific reason we're defining it here, rather than the `constructor`?

Reply:
Yeah we reinit there its fine. Not asking to remove from `flattenSection`, instead asking to define the member variable with default value in the constructor.

Reply:
It can be defined in constructor but need to empty array at the entry of `flattenSections` in the case when called from `componentDidUpdate` because items already added in constructor